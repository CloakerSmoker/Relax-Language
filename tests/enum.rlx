
enum animal_kind {
	cat
	dog
	frog
	fish
}

enum starts_from_10 {
	first := 10
	second
	third
}

enum sparse {
	thirty := 30
	fourty := 40
	fifty := 50
}

#Require "$/Console.rlx"

define void PrintAnimalType(animal_kind kind) {
	i8*[4] Names {"cat", "dog", "frog", "fish"}
	
	Print("%s", Names[kind As i32])
}

define void PrintAnimalType2(animal_kind kind) {
	Print("%s", nameof:(kind))
}

define void PrintSparseName(sparse s) {
	Print("%s", nameof:(s))
}

define i32 Main() {
	PrintAnimalType(animal_kind:dog)
	PrintAnimalType(animal_kind::cat)
	
	Print("%i", starts_from_10::second)

	PrintAnimalType(ANIMAL_KIND_DOG)

	PrintAnimalType2(ANIMAL_KIND_CAT)
	PrintAnimalType2(ANIMAL_KIND_FISH)
	PrintAnimalType2(20 As animal_kind)

	PrintSparseName(0 As sparse)
	PrintSparseName(sparse::thirty)
	PrintSparseName(sparse::fourty)
	PrintSparseName(sparse::fifty)
	PrintSparseName(60 As sparse)
	
	return 0
}