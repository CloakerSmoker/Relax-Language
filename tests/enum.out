dogcat11dogcatfish<%s null><%s null>thirtyfourtyfifty<%s null>